Add `CONTRIBUTING.md` (#366)
canonical/charmed-openstack-upgrader: CONTRIBUTING.md

# Contributing to Charmed OpenStack Upgrader (COU)

> Thank you for your interest in helping us with improving **COU**! We're open to

Reviewer: ```suggestion
Thank you for your interest in helping us improve **COU**! We're open to
```

  key, typically using GPG or SSH, and register it in your GitHub account to
  verify the origin of your code changes. For instructions on setting this up,
  please refer to
  [Managing commit signature verification](https://docs.github.com/en/authentication/managing-commit-signature-verification).

## Code contribution
If you're interested in making code contributions, please begin by forking the
repository to your own GitHub account. From there, you can open Pull Requests (PRs)
against the `main` branch of the upstream repository.

Please adhere to the following guidelines prior to submitting your changes: 

- Add or update any unit tests accordingly if applicable.
- Format code with `make reformat` (which runs `black` and `isort`)
- Ensure unit tests and/or linting checks pass by running `make lint` and
  `make unittests`
  - For documentation contribution, run the following commands and confirm that
  - no errors are raised:
    ```bash
    cd docs/
    make clean
    make html
    make spelling
    make woke
    make linkcheck
    ``` 
- Commit messages should be well-structured and provide a meaningful explanation

> - of the changes made

Reviewer: ```suggestion
  of the changes made
```

- Commits must be signed (refer to the [Prerequisites](#prerequisites) section)

## Find issues to work on
[GitHub Issues](https://github.com/canonical/charmed-openstack-upgrader/issues)
is our central hub for bug tracking and issue management, with labels used to
organize them into different categories. For new contributors, we recommend
starting with issues labeled "good first issue." If you're interested in
enhancing our documentation, you can filter issues using the "documentation"
label to find issues specifically related to documentation improvement.
  

> Once you decided which issue to work on, you can express your interest by posting

Reviewer: ```suggestion
Once you have decided which issue to work on, you can express your interest by posting
```
